### DISCUSSION TITLE
유니티 테스트코드 관련 질문

### DISCUSSION THREAD

**Discussion by fkdl0048** [fkdl0048/Mentoring-Project#107] [Q&A] [answered]
유니티로 테스트코드를 짜는데 TestRunner를 통해 짤 때 말씀해주신 Unit단위 테스트만 할 땐 Editor단위에서 테스트 코드를 돌려도 될 것 같다는 생각이 들어서 질문합니다.

Play레벨은 코루틴으로 실제 런타임에서 검사해야하는 테스트라면, 지금 필요한 테스트는 간단한 테스트라 Editor위치에 있어도 될 것 같다는 생각이..

실제로는 테스트를 어떻게 분리하나요?

![image](https://github.com/fkdl0048/Mentoring-Project/assets/84510455/f7a6adfd-f6ba-44db-9e2d-20df34922a86)

지금은 인게임에서 사용하는 스크립트는 Scripts 어셈블리로 묶어두고, 테스트 코드는 Runtime과 Editor로 구분하여 각각 묶어두었습니다

테스트 코드의 성격과 유니티에서 요구하는 Editor와 Runtime이 일치 해야하는지 궁금합니다!

**Comment by jongfeel** [comment, marked as answer]
이것도 지난 번 미팅 때 말씀 드린 건데

EditorMode test는 Editor용 코드 테스트할 때 쓰는 거고
일반적으로는 PlayMode 테스트를 사용하는게 맞습니다.
